script.js

```diff
@@ -0,0 +1,112 @@
+const game = { turn: 0 };
+const players = [{ score: 0 }, { score: 0 }];
+const diagonals = [
+  [
+    { row: 0, column: 0 },
+    { row: 1, column: 1 },
+    { row: 2, column: 2 },
+  ],
+  [
+    { row: 0, column: 2 },
+    { row: 1, column: 1 },
+    { row: 2, column: 0 },
+  ],
+];
+let cells;
+
+function getCellCoordinates(cell) {
+  const index = cells.indexOf(cell);
+  return { row: Math.floor(index / 3), column: index % 3 };
+}
+
+function initializeBoard() {
+  game.board = [
+    [null, null, null],
+    [null, null, null],
+    [null, null, null],
+  ];
+  registerCellClickListeners();
+}
+
+function updateBoard(coordinates) {
+  game.board[coordinates.row][coordinates.column] = game.turn;
+  game.changedCoordinates = coordinates;
+}
+
+function toggleTurn() {
+  game.turn = game.turn === 0 ? 1 : 0;
+}
+
+function recordVictory() {
+  players[game.turn].score += 1;
+}
+
+function getGameFirstTurnPlayer() {
+  return (players[0].score + players[1].score) % 2;
+}
+
+function startNextGame() {
+  initializeBoard();
+  game.turn = getGameFirstTurnPlayer();
+}
+
+function allEqual(array) {
+  return array.every((value) => value === array[0]);
+}
+
+function includes(coordinatesList, coordinates) {
+  return (
+    coordinatesList.filter(
+      (x) => x.row === coordinates.row && x.column === coordinates.column
+    ).length > 0
+  );
+}
+
+function getTriples(coordinates) {
+  const rowTriple = game.board[coordinates.row];
+  const columnTriple = game.board.map((r) => r[coordinates.column]);
+  const diagonalTriples = diagonals
+    .filter((diagonal) => includes(diagonal, coordinates))
```

Comment:
I was confused by this for a while because of the JavaScript `Array.includes()` function. Something that also contributed to my confusion was the order of parameters. JavaScript's includes function has as its first parameter: `valueToFind`. The equivalent parameter here happens to come in the second position.

Reply:
My rationale for the parameter order is that the array and value remain in the same order as in a call to the built-in function. For example:
- `myArray.includes(myValue)`
- `includes(myArray, myValue)`

Would you prefer a different function name? We could use something like `coordinatesListIncludes`, though that is a bit redundant depending on one's perspective.

Reply:
Yeah that explains the order, I'm fine to just leave this unchanged